### ISSUE TITLE
addSearchPath duplicate not handled

### ISSUE THREAD

**Issue by imtrobin** [cocos2d/cocos2d-x#20302]
I noticed if you add same search path, it does not check against duplicate

```
FileUtils::getInstance ()->addSearchPath ("test");
FileUtils::getInstance ()->addSearchPath ("test");

int s = FileUtils::getInstance ()->getSearchPaths ().size (); // keeps increasing
```
I see there is PR in 3.8 beta to check for duplicate before insert, but I don't see that patch in current addSearchPath

https://github.com/cocos2d/cocos2d-x/commit/22b4f5a11eaa142510c04adc553cb2bafb8996b1

**Comment by minggo** [comment]
Yep, i don't know why. It seems the log doesn't include the commit. Will you please send a PR to fix it? Thanks.
